### DISCUSSION TITLE
Disable Flatlist scrolling when GestureHandler interacting?

### DISCUSSION THREAD

**Discussion by CallumHemsley** [software-mansion/react-native-gesture-handler#1667] [Q&A]
Hello, I'm currently trying to wrap a `PanGestureHandler`  around a react-native flatlist.

I've got the pangesture to work with the flatlist by giving simultaneous handlers and wrapping the flatlist with a `NativeViewGestureHandler`.

However the scrolling is still enabled on the FlatList. I've also tried giving the NativeViewGestureHandler a `waitFor` which works until the FlatList re-renders, afterwards not working as expected.

I've also tried giving the Flatlist a `useState` boolean variable to enable/disable the scrolling however this doesn't update fast enough so it will scroll a little bit before being disabled.

Does anyone know a solution to this? I really want the PanGestureHandler to wrap the flatlist so that the flatlist can re-render and update its items without making the PanGestureHandler re-render and therefore stop it's interaction logic.

**Comment by JasonCareter** [comment]
you may use the ref to disable scrolling. add ref to the scrollable component.
just call the function showed below when you're trying to disable/enable scrolling
```typescript
vref.current && vref.current.setNativeProps({scrollEnabled: false/true })
```

**Reply by GeorgeHop** [reply]
Probably this will not soo smoth as in this videos https://github.com/software-mansion/react-native-gesture-handler/discussions/2316

**Comment by GeorgeHop** [comment]
Any success? Because I also trying to achieve animation like that https://github.com/software-mansion/react-native-gesture-handler/discussions/2316

**Reply by GeorgeHop** [reply]
And also I found this thread https://github.com/software-mansion/react-native-gesture-handler/discussions/1836
